pkg/ruler/notifier.go

```diff
@@ -16,8 +17,21 @@ import (
 	"github.com/prometheus/prometheus/discovery"
 	"github.com/prometheus/prometheus/discovery/dns"
 	"github.com/prometheus/prometheus/notifier"
+
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/cortexproject/cortex/pkg/util/tls"
 )
 
+type NotifierConfig struct {
+	TLS       tls.ClientConfig `yaml:"tls"`
```

Comment:
We've just revived https://github.com/cortexproject/cortex/pull/3156 where we added an explicit flag to enable TLS. I would suggest to keep the same pattern as `pkg/util/grpcclient/grpcclient.go` here, so:
1. Add `TLSEnabled bool`
2. Make `TLS` inline in the YAML

Reply:
In #3156 we are not adding TLSEnabled for HTTP clients, only to gRPC.

Reply:
Oh right, because it's the HTTP URL scheme to tell if it's TLS or not. Then ignore this comment.